src/components/Reporting.js

```diff
@@ -15,36 +22,55 @@ class Reporting extends Component {
   constructor(props) {
     super(props);
     this.setActiveTab = this.setActiveTab.bind(this);
+    this.renderActiveTab = this.renderActiveTab.bind(this);
     this.state = {
       activeTab: 'Dashboard'
     };
   }
 
+  componentWillMount() {
+    this.props.loadTransactions();
+  }
+
   setActiveTab(tabName) {
     return () => this.setState({ activeTab: tabName });
   }
 
-  componentWillMount() {
-    this.props.loadTransactions();
+  renderActiveTab() {
+    const activeTabComponent = this.props.tabs[this.state.activeTab];
+    return activeTabComponent({ transactions: this.props.transactions });
   }
 
   render() {
-    return ReportingNav(
-      {
-        activeTab: this.state.activeTab,
-        setActiveTab: this.setActiveTab
-      }
+    return (
+      dom.div(
+        null,
+        ReportingNav(
+          {
+            activeTab: this.state.activeTab,
+            tabNames: _.keys(this.props.tabs),
+            setActiveTab: this.setActiveTab
+          }
+        ),
+        this.renderActiveTab()
+      )
     );
   }
 }
 
 Reporting.propTypes = {
   loadTransactions: PropTypes.func.isRequired,
-  transactions: PropTypes.array
+  transactions: PropTypes.array,
+  tabs: PropTypes.object
 };
 
 Reporting.defaultProps = {
-  transactions: []
+  transactions: [],
+  tabs: {
```

Comment:
I think this is fine, since we are only rendering the active tab. Another way would be to render all tabs and change which one is active using bootstrap's tab component.